RELEASE/scripts/autoscend/auto_util.ash

```diff
@@ -1961,9 +1797,83 @@ boolean auto_deleteMail(kmailObject msg)
 	return false;
 }
 
+boolean LX_summonMonster()
+{
+	// get war outfit if have yr available
+	if(!possessOutfit("Frat Warrior Fatigues") && auto_warSide() == "fratboy" && adjustForYellowRayIfPossible())
+	{
+		// attempt to use calculate the universe
+		if(summonMonster($monster[War Frat 151st Infantryman])) return true;
+		// attempt to summon other sources of outfit
+		if(summonMonster($monster[War Frat Mobile Grill Unit])) return true;
+		if(summonMonster($monster[orcish frat boy spy])) return true;
+	}
+	if(!possessOutfit("War Hippy Fatigues") && auto_warSide() == "hippy" && adjustForYellowRayIfPossible())
+	{
+		if(summonMonster($monster[War Hippy Airborne Commander])) return true;
+		if(summonMonster($monster[war hippy spy])) return true;
+	}
+
+	// summon mountain man if we know the ore we need and still need 2 or more
+	item oreGoal = to_item(get_property("trapperOre"));
+	if(internalQuestStatus("questL08Trapper") < 2 && oreGoal != $item[none] && item_amount(oreGoal) < 2 && adjustForYellowRayIfPossible())
+	{
+		if(summonMonster($monster[mountain man])) return true;
+	}
+
+	// only summon NSA if in hardcore as we will pull items in normal runs
+	if(internalQuestStatus("questL08Trapper") < 3 && in_hardcore() && my_level() >= 8 && !get_property("auto_L8_extremeInstead").to_boolean())
+	{
+		boolean wantSummonNSA = item_amount($item[ninja rope]) < 1 || 
+			item_amount($item[ninja carabiner]) < 1 || 
+			item_amount($item[ninja crampons]) < 1;
+		if(wantSummonNSA)
+		{
+			if(summonMonster($monster[Ninja Snowman Assassin])) return true;
+		}
+	}
+
+	// summon ghost for white pixels if at tower door and still need digital key
+	if(needDigitalKey() && !auto_hasPowerfulGlove() && internalQuestStatus("questL13Final") == 5)
+	{
+		handleFamiliar("item");
+		addToMaximize("20 item");
+		equipMaximizedGear();
+		if(summonMonster($monster[Ghost])) return true;
+	}
+
+	// summon LFM if don't have autumnaton since that guarantees 1 turn to get 5 barrels
+	if(item_amount($item[barrel of gunpowder]) < 5 && get_property("sidequestLighthouseCompleted") == "none" && 
+	my_level() >= 12 && !auto_hasAutumnaton())
+	{
+		if(summonMonster($monster[lobsterfrogman])) return true;
+	}
+
+	if(auto_is_valid($item[Smut Orc Keepsake Box]) && item_amount($item[Smut Orc Keepsake Box]) == 0 && (lumberCount() < 30 || fastenerCount() < 30))
```

Comment:
suggest adding a condition like
>(my_level() >= 9 || my_adventures() < 7+ auto_advToReserve())

to leave an opportunity to get the more valuable level 8 mountain man first if cargo summon is its only available source?
!(auto_have_skill($skill[Stuffed Mortar Shell]) && auto_have_skill($skill[Spirit of Peppermint])) could be added as another rough condition for not waiting for mountain man, ores being more valuable than keepsake box is assuming blech house every 3 fights

Reply:
Good point. Added the level check both for this reason but also so we know we can beat the monster as no yellow ray is used